### DISCUSSION TITLE
Go Version Updating Patch

### DISCUSSION THREAD

**Discussion by owenhaynes** [renovatebot/renovate#23824] [Suggest an Idea]
### Tell us more.

Since go 1.21.0 its possible to use patch versions in go.mod files. It would be nice if the version bumper would be able to increase the patch version of Go in the module file.

The Go Toolchain now auto downloads forward and backwards the toolchain from the version defined in `go.mod`. So being able to auto update the patch version means we can make sure people are using the correct version of Go, when building on CI and locally

This may need to be put behind a feature flag as patch versions in `go.mod` do not work in go versions < 1.21.

https://go.dev/doc/toolchain

**Comment by rarkins** [comment]
Do you mean updating the go mod directive, which I think is disabled by default today?

**Reply by owenhaynes** [reply]
Yes I enabled the feature using 
```
  "packageRules": [
    {
      "matchDatasources": ["golang-version"],
      "rangeStrategy": "bump"
    }
  ]
 ```
Which is taken from https://github.com/renovatebot/renovate/blob/e61e3c80bcd00dbec0651d2b4cf097e6c0e85317/lib/modules/versioning/go-mod-directive/readme.md
 
This currently only bumps the minor versions but from go 1.21.0 patch versions can be used in the go mod directive so this should be improved to allow that.

**Reply by rarkins** [reply]
If there's a bug or limitation please create a reproduction repo to demonstrate
